stdlib/public/Concurrency/GlobalExecutor.cpp

```diff
@@ -448,13 +448,9 @@ void swift::swift_task_enqueueOnDispatchQueue(Job *job,
 }
 #endif
 
-#if SWIFT_CONCURRENCY_COOPERATIVE_GLOBAL_EXECUTOR
-static HeapObject _swift_mainExecutorIdentity;
-#endif
-
 ExecutorRef swift::swift_task_getMainExecutor() {
 #if SWIFT_CONCURRENCY_COOPERATIVE_GLOBAL_EXECUTOR
-  return ExecutorRef::forOrdinary(&_swift_mainExecutorIdentity, nullptr);
+  return ExecutorRef::generic();
```

Comment:
This is interesting. In the cooperative case, does`@MainActor` not really have its own executor, because all executors are the same? Or is it that we should have a main-actor actor and dispatching to it goes in through the same run loop? I think I agree that this is the right approach.

Reply:
Good point. I respected the former from my understanding, but I'm not confident here.